sklearn/multioutput.py

```diff
@@ -162,10 +162,11 @@ def partial_fit(self, X, y, classes=None, sample_weight=None, **partial_fit_para
             )
 
         if _routing_enabled():
+            if sample_weight is not None:
```

Comment:
So now, we have the same pattern than in the case of when `_routing_enabled is False` which make sense.

I think that we have the patter of calling `process_routing` and passing `sample_weight` in a lot of place if I am not wrong. Would it make sense to do something special in `process_routing` specifically if `sample_weight is None` instead of handling it in the meta-estimator itself?

Reply:
This we cannot do since sometimes passing `None` explicitly does something, like silencing a deprecation warning.